[ESP32] error: 'analogWrite' was not declared in this scope

When trying to compile this library for ESP32 using PlatformIO the following error occurs:

```
.pio/libdeps/lolin32/Analog-Digital Multiplexers_ID5363/src/Mux.cpp: In member function 'int8_t Mux::write(uint8_t, int8_t)':
.pio/libdeps/lolin32/Analog-Digital Multiplexers_ID5363/src/Mux.cpp:121:32: error: 'analogWrite' was not declared in this scope
     analogWrite(signalPin, data);
                                ^
```

platformio config:

```ini
[env:lolin32]
platform = espressif32
board = lolin32
framework = arduino

lib_deps =
  Analog-Digital Multiplexers
```

It also redefines ANALOG and DISABLED..

```
.pio/libdeps/lolin32/Analog-Digital Multiplexers_ID5363/src/Defs.h:22:0: warning: "DISABLED" redefined
 #define DISABLED 0
 ^
In file included from /home/foo/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:53:0,
                 from /home/foo/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:35,
                 from .pio/libdeps/lolin32/Analog-Digital Multiplexers_ID5363/src/Mux.h:20,
                 from .pio/libdeps/lolin32/Analog-Digital Multiplexers_ID5363/src/Mux.cpp:17:
/home/foo/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:51:0: note: this is the location of the previous definition
 #define DISABLED  0x00
 ^

.pio/libdeps/lolin32/Analog-Digital Multiplexers_ID5363/src/Defs.h:19:0: warning: "ANALOG" redefined
 #define ANALOG 0
 ^
In file included from /home/foo/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:53:0,
                 from /home/foo/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:35,
                 from .pio/libdeps/lolin32/Analog-Digital Multiplexers_ID5363/src/Mux.h:20,
                 from .pio/libdeps/lolin32/Analog-Digital Multiplexers_ID5363/src/Mux.cpp:17:
/home/foo/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:48:0: note: this is the location of the previous definition
 #define ANALOG            0xC0
 ^
```

--- a/src/Mux.cpp
+++ b/src/Mux.cpp
@@ -118,7 +118,11 @@ int8_t Mux::write(uint8_t data, int8_t channel) {
 
   switch (signalType) {
   case ANALOG:
-    analogWrite(signalPin, data);
+    #if defined(ARDUINO_ARCH_ESP32)
+	    ledcWrite(signalPin, data);
+	#else
+		analogWrite(signalPin, data);
+	#endif
     break;
   case DIGITAL:
     digitalWrite(signalPin, data);

--- a/src/Mux.cpp
+++ b/src/Mux.cpp
@@ -119,10 +119,10 @@ int8_t Mux::write(uint8_t data, int8_t channel) {
   switch (signalType) {
   case ANALOG:
     #if defined(ARDUINO_ARCH_ESP32)
-	    ledcWrite(signalPin, data);
-	#else
-		analogWrite(signalPin, data);
-	#endif
+    ledcWrite(signalPin, data);
+    #else
+    analogWrite(signalPin, data);
+    #endif
     break;
   case DIGITAL:
     digitalWrite(signalPin, data);

